shell/common/native_mate_converters/value_converter.h

```diff
@@ -41,10 +41,12 @@ struct Converter<base::Optional<T>> {
                      v8::Local<v8::Value> val,
                      base::Optional<T>* out) {
     if (val->IsNull() || val->IsUndefined()) {
+      *out = base::nullopt;
       return true;
     }
     T converted;
-    if (Converter<T>::FromV8(isolate, val, &converted)) {
+    if (!Converter<T>::FromV8(isolate, val, &converted)) {
+      *out = base::nullopt;
```

Comment:
It isn't a failed conversion because this is an optional param.  If this param fails to convert the optional will be empty and the next arg will use this param.

This emulates expected behavior of optional arguments in the middle of argument lists.